docs/spec/expressions.md

```diff
@@ -106,6 +106,27 @@ Given the above declaration, the expression `x.y.z` would evaluate to the litera
 
 Property accessors can be used with any object. This includes parameters and variables of object types and object literals. Using a property accessor on an expression of non-object type is an error.
 
+## Nested resource accessors
+Nested resource accessors are used to access resources that are declared inside another resource. Only top-level resources are considered top-level declarations.  
```

Comment:
This could do with a bit of clarification before showing the example.
* Mention they act similarly to property access but for nested resources? (to give beginners something to latch onto)
* "top-level declarations" may not mean a whole lot to people - what about something like "Nested resources cannot be referenced using their symbolic name outside of the parent resource body. Instead they must be qualified with the parent symbolic name using the `:` resource accessor."?
* Mention the specific `:` token used for the syntax.
* Add a note about how the operator works  with scoping for accessing sibling resources?
* May be worth mentioning the semantics about cycles with parent -> child properties as the reason why might not be obvious to users.

Reply:
Agree with the above. Especially the last point. Might also be worth to explain the need for parentheses when using the ternary somewhere in the docs for conditions.

Reply:
I'll add info about cycles to the other parts of the docs about nested resources. this is about the `:` operator which is not a factor related to cycles.